integration-testing/build.gradle

```diff
@@ -43,70 +62,34 @@ compileDebugAgentTestKotlin {
     }
 }
 
-task npmTest(type: Test) {
-    def sourceSet = sourceSets.npmTest
-    environment "projectRoot", project.rootDir
-    environment "deployVersion", version
-    def dryRunNpm = project.properties['dryRun']
-    def doRun = dryRunNpm == "true" // so that we don't accidentally publish anything, especially before the test
-    onlyIf { doRun }
-    if (doRun) { // `onlyIf` only affects execution of the task, not the dependency subtree
-        dependsOn(project(':').getTasksByName("publishNpm", true))
-    }
-    testClassesDirs = sourceSet.output.classesDirs
-    classpath = sourceSet.runtimeClasspath
-}
-
 task mavenTest(type: Test) {
-    environment "version", version
+    environment "version", coroutines_version
     def sourceSet = sourceSets.mavenTest
-    dependsOn(project(':').getTasksByName("publishToMavenLocal", true))
     testClassesDirs = sourceSet.output.classesDirs
     classpath = sourceSet.runtimeClasspath
-    // we can't depend on the subprojects because we need to test the classfiles that are published in the end.
-    // also, we can't put this in the `dependencies` block because the resolution would happen before publication.
-    def mavenTestClasspathConfiguration = project.configurations.detachedConfiguration(
-            project.dependencies.create("org.jetbrains.kotlinx:kotlinx-coroutines-core:$version"),
-            project.dependencies.create("org.jetbrains.kotlinx:kotlinx-coroutines-android:$version"))
-
-    mavenTestClasspathConfiguration.attributes {
-        attribute(KotlinPlatformType.attribute, KotlinPlatformType.jvm)
-    }
-
-    classpath += mavenTestClasspathConfiguration
 }
 
 task debugAgentTest(type: Test) {
     def sourceSet = sourceSets.debugAgentTest
-    dependsOn(project(':kotlinx-coroutines-debug').shadowJar)
-    jvmArgs ('-javaagent:' + project(':kotlinx-coroutines-debug').shadowJar.outputs.files.getFiles()[0])
+    def coroutinesDebugJar = sourceSet.runtimeClasspath.filter {it.name == "kotlinx-coroutines-debug-${coroutines_version}.jar" }.singleFile
+    jvmArgs ('-javaagent:' + coroutinesDebugJar)
     testClassesDirs = sourceSet.output.classesDirs
     classpath = sourceSet.runtimeClasspath
     systemProperties project.properties.subMap(["overwrite.probes"])
```

Comment:
Now, that's an issue.

This flag control the functionality in `integration-testing/src/debugAgentTest/kotlin/PrecompiledDebugProbesTest.kt` that allows us to replace `DebugProbesKt.bin`, a file in the repository, with a new version. I'm sure this functionality is now broken. We should either fix this or adapt in some other way. @qwwdfsad, do we still need `-Poverwrite.probes`? Does anyone rely but us on this, or can we provide a more fitting interface? Also, is the comment below still relevant (from the file with the test)?
> ensure that classfile has major version equal to 50 (Java 6 compliance)

Reply:
The comment is indeed redundant.

>We should either fix this or adapt in some other way

Right. @mvicsokolova please ensure that ` ./gradlew debugAgentTest -Poverwrite.probes=true` properly works. For now it fails with `FileNotFoundException`

Reply:
Apart from that, `./gradlew publishToMavenLocal` + `cd integration-testing` + ` ./gradlew debugAgentTest` fails locally at `PrecompiledDebugProbesTest`. It shouldn't be be the case, please investigate